### PR TITLE
handle ttl=0

### PR DESCRIPTION
**Pull request by ekmutai**
- Fixes: ERR invalid expire time in set when ttl is set to 0
- Allow passing of ttl=0 to remove expiry

### CODE REVIEW

**Comment by botika** [comment]
Please put test for this bug. Thanks!

**Comment by ekmutai** [comment]
Done @botika

File: src/index.ts [hunk 1]
```diff
@@ -68,11 +68,11 @@ function builder<T extends Clients>(
       if (!isCacheable(value))
         throw new NoCacheableError(`"${value}" is not a cacheable value`);
       const t = ttl === undefined ? options?.ttl : ttl;
-      const ttlOptions = t !== undefined ? { PX: t } : undefined;
+      const ttlOptions = t !== undefined && t !== 0 ? { PX: t } : undefined;
       await redisCache.set(key, getVal(value), ttlOptions);
     },
     async mset(args, ttl) {
-      const t = ttl === undefined ? options?.ttl : ttl;
+      const t = ttl === undefined || ttl === 0 ? options?.ttl : ttl;
```
Comment (botika) [inline src/index.ts:75]: bad

File: test/index.test.ts [hunk 2]
```diff
@@ -58,6 +58,11 @@ describe('set', () => {
     await expect(redisCache.get('foo')).resolves.toBe('bar');
   });
 
+  it('should store a value with a ttl of zero', async () => {
+    await expect(redisCache.set('foo', 'bar', 0)).resolves.toBeUndefined();
+    await expect(redisCache.get('foo')).resolves.toBe('bar');
+  });
+
```
Comment (botika) [inline test/index.test.ts:65]: uncompleted

File: test/index.test.ts [hunk 3]
```diff
@@ -105,6 +110,20 @@ describe('mset', () => {
...
+        ['foo2', 'bar2'],
+      ],
+      0,
+    );
+    await expect(redisCache.store.mget('foo', 'foo2')).resolves.toStrictEqual([
+      'bar',
+      'bar2',
+    ]);
+  });
+
```
Comment (botika) [inline test/index.test.ts:126]: uncompleted

**Review by botika — CHANGES_REQUESTED** [review]

**Comment by ekmutai** [comment]
@botika please review and share any further feedback

**Comment by botika** [comment]
Sorry, I did it. #156

**Pull request closed without merging**